src/std_io.py

```diff
@@ -0,0 +1,167 @@
+                # airgap mode uses the timestamp
+                # unhex the timestamp with error rejection
+                if len(hexTStamp) == 16:
+                    try:
+                        # stdioStamp, = unpack('>Q', unhexlify(hexTStamp))
+                        _, = unpack('>Q', unhexlify(hexTStamp))
+                    except Exception:
+                        logger.info("STDIN: Invalid timestamp format: " + hexTStamp)
+                        continue
+
+            # check that proof of work is sufficient.
+            if not protocol.isProofOfWorkSufficient(binObject):
+                logger.info('STDIN: Insufficient Proof of Work')
+                continue
+
+            # extract expiry time, object type
+            eTime, = unpack('>Q', binObject[8:16])
+            objectType, = unpack('>I', binObject[16:20])
+
+            # extract version number and stream number
+            readPosition = 20  # bypass the nonce, time, and object type
+            # versionNumber, versionLength
+            _, versionLength = decodeVarint(binObject[readPosition:readPosition + 10])
+            readPosition += versionLength
+            streamNumber, streamNumberLength = decodeVarint(binObject[readPosition:readPosition + 10])
+            readPosition += streamNumberLength
+
+            # calculate inventory hash
+            inventoryHash = calculateInventoryHash(binObject)
+
+            # duplicate check on inventory hash (both netcat and airgap)
+            if inventoryHash in Inventory():
+                logger.info("STDIN: Already got object " + hexlify(inventoryHash))
+                continue
+
+            # in netcat mode, push object to inventory and id to output queue
+            if stdInputMode == 'netcat':
+                Inventory()[inventoryHash] = (objectType, streamNumber, binObject, eTime, '')
+                logger.info("STDIN: Accepted object (type=%u) " % objectType + hexlify(inventoryHash))
+                queues.invQueue.put((streamNumber, inventoryHash))
+
+            # honour global shutdown flag
+            if state.shutdown != 0:
+                logger.info('stdInput thread shutdown.')
+                break
+
+
+class objectStdOut(threading.Thread):
+    """
+    The objectStdOut thread receives network objects from the receiveDataThreads.
+    """
+    def __init__(self):
+        threading.Thread.__init__(self, name="objectStdOut")
+
+        # REFACTOR THIS with objectProcessor into objectProcessorQueue
+        queryreturn = sqlQuery(
+            '''SELECT objecttype, data FROM objectprocessorqueue''')
+        for row in queryreturn:
+            objectType, data = row
+            queues.objectProcessorQueue.put((objectType, data))
+        sqlExecute('''DELETE FROM objectprocessorqueue''')
+        logger.debug('Loaded %s objects from disk into the objectProcessorQueue.' % str(len(queryreturn)))
+        # /REFACTOR THIS
+
+    def run(self):
+        while True:
```

Comment:
When the queue is empty would this not cause unnecessary 100% CPU? Perhaps a small sleep is needed inside the while loop?